src/settings.py

```diff
@@ -44,6 +43,8 @@
 KILL_IDLE_TIME = 300
 WARN_IDLE_TIME = 180
 PM_WARN_IDLE_TIME = 240
+JOIN_PHASE_KILL_IDLE_TIME = 1800
+JOIN_PHASE_WARN_IDLE_TIME = 1500
```

Comment:
way too long imo, I'd warn after 15 minutes and kill after 20 at the most; this really needs to be discussed though -- feel free to prod -ops to kick that off.